misc-utils/lsfd-file.c

```diff
@@ -112,6 +112,39 @@ static uint64_t get_map_length(struct file *file)
 	return res;
 }
 
+int decode_source(char *buf, size_t bufsize,
+		  unsigned int dev_major, unsigned int dev_minor,
+		  enum decode_source_level level)
+{
+	if (bufsize == 0)
+		return 0;
+
+	buf[0] = '\0';
+
+	if (level & DECODE_SOURCE_FILESYS_BIT) {
+		if (dev_major == 0) {
+			const char *filesystem = get_nodev_filesystem(dev_minor);
+			if (filesystem)
+				return snprintf(buf, bufsize, "%s", filesystem);
```

Comment:
It would probably be more effective to use `xstrncpy(buf, filesystem, bufsiz)` than `snprinf("%s")`. The decode_source() return code is nowhere used, so 0/-1  or void would be also good enough.

Reply:
Thank you. Updated.